### PR TITLE
INTYGFV-15168: Added TsDiabetes to allowedCertificateTypes & refactored SendCertificateFunctionImpl to minimize excessive if-statements.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java [hunk 1]
```diff
@@ -70,55 +73,75 @@ public class SendCertificateFunctionImpl implements SendCertificateFunction {
...
+    private static boolean isWrongMajorVersion(Certificate certificate) {
+        return (certificate.getMetadata().getType().equals(TsBasEntryPoint.MODULE_ID)
+            || certificate.getMetadata().getType().equals(TsDiabetesEntryPoint.MODULE_ID)) && !certificate.getMetadata()
+            .isLatestMajorVersion();
+    }
+
+    private Map<String, Optional<ResourceLinkDTO>> getResourceLinkDTO(Certificate certificate) {
+        return Map.of(
+            LisjpEntryPoint.MODULE_ID, getResourceLinkDTOFkLisjp(certificate),
+            LuseEntryPoint.MODULE_ID, getResourceLinkDTOFk(),
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java:99]: Suggesting a switch-case here instead. Seems that only one resource link is fetched on each call, no need to prepare all of them each time(?)

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java [hunk 2]
```diff
@@ -70,55 +73,75 @@ public class SendCertificateFunctionImpl implements SendCertificateFunction {
...
+                SEND_TO_FK,
+                SEND_TO_FK_DESCRIPTION,
+                SEND_BODY_FK,
+                true));
     }
 
     private static boolean shouldSendToFk(Certificate certificate) {
         return certificate.getMetadata().getType().equalsIgnoreCase(LuaenaEntryPoint.MODULE_ID) || certificate.getMetadata().getType()
-            .equalsIgnoreCase(LuseEntryPoint.MODULE_ID) || certificate.getMetadata().getType().equalsIgnoreCase(LuaefsEntryPoint.MODULE_ID);
+            .equalsIgnoreCase(LuseEntryPoint.MODULE_ID) || certificate.getMetadata().getType().equalsIgnoreCase(LuaefsEntryPoint.MODULE_ID)
```
Comment (mikeyOne) [inline web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java:143]: Is this method still used?

**Review by mikeyOne — COMMENTED** [review]
Only 2 questions, see below

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java:99]: Thats a great suggestion, will do!

File: web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java:143]: No, this function can we removed. Done.

**Review by mikeyOne — APPROVED** [review]

**Pull request merged**